### PR TITLE
template the kustomization.yaml file 

### PR DESCRIPTION
**Pull request by dcmcand**
## Reference Issues or PRs
part of #2557 

## What does this implement/fix?
Adds the ability to template the `kustomization.yaml` file to allow variables from the config to be dynamically set.

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [X] Did you test the pull request locally?
- [ ] Did you add new tests?

## How to test this PR?
1. enable healthchecks with
```yaml
monitoring:
  healthchecks:
    enabled: true:
```
2. run a render with `nebari render -c nebari-config.yaml`
3. check the rendered objects namespace with `grep -r  namespace stages/10-kubernetes-kuberhealthy/` and `grep -r  namespace stages/11-kubernetes-kuberhealthy-healthchecks/` and verify that the correct namespace has been set (it should be `dev`) by default.
4. change the namespace to a different namespace in your `nebari-config.yaml` file and rerun the render
5. using the above commands, verify the namespace was updated.
6. return to your original namespace and rerun the render
7. run a `nebari deploy` to validate that everything deploys to the correct namespace

### CODE REVIEW

**Comment by kcpevey** [comment]
Reminder to add documentation for this feature - we can add a "this is in beta testing" header

**Review by aktech — CHANGES_REQUESTED** [review]

File: src/_nebari/stages/base.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import tempfile
 from typing import Any, Dict, List, Tuple
 
+from jinja2 import Environment, FileSystemLoader
```
[the remark this answers is not in the document]
Comment (dcmcand) [reply src/_nebari/stages/base.py:10]: Good catch, thanks

File: src/_nebari/stages/kubernetes_kuberhealthy/template/kustomization.yaml.tmpl [hunk 2]
```diff
@@ -6,6 +6,10 @@ helmCharts:
 - name: kuberhealthy
   repo: https://kuberhealthy.github.io/kuberhealthy/helm-repos
   releaseName: kuberhealthy
-  namespace: dev
-  version: "100" # v.2.7.1
+  namespace: {{ namespace }}
+  version: "{{ kuberhealthy_version }}" # v.2.7.1
```
[the remark this answers is not in the document]
Comment (dcmcand) [reply src/_nebari/stages/kubernetes_kuberhealthy/template/kustomization.yaml.tmpl:10]: It is indeed. Thanks!

**Comment by dcmcand** [comment]
Thanks @aktech, I addressed all of your comments here.

**Review by aktech — APPROVED** [review]
Looks great to me, thanks! 🎉  This is good to merge!

**Pull request merged**